accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -2131,6 +2162,55 @@ If `p` with required type `T` is in an irrefutable context:
     a _cast_ from `dynamic` to `String` and instead let the `String s` pattern
     _test_ the value's type, which then fails to match.*
 
+*   Next, it is a compile-time error if `M` is not assignable to `T`.
+    *Destructuring, variable, and identifier patterns can only be used in
+    declarations and assignments if we can statically tell that the
+    destructuring and variable binding won't fail to match (though it may still
+    throw at runtime if the matched value type is `dynamic`).*
+
+### Coercions
+
+The language specification documents do not yet define a unified concept of
+_coercions_, and they do not define what it means to _attempt to insert a
+coercion_. However, the following is intended to establish these concepts
+in a sufficiently precise manner to enable the implementation of patterns:
+
+The language supports the following mechanisms, which are the currently
+existing _coercions_:
+
+- Implicit generic function instantiation.
+- Implicit tear-off of a `.call` method.
+- Implicit tear-off of a `.call` method, which is then generically instantiated.
```

Comment:
(I'd still like to include implicit downcasts, even if I know that it differs in not actually changing the runtime type of the value.)

Reply:
(I just want more clarity about what a coercion is, and how we treat them. ;-)